gordo/client/cli/client.py

```diff
@@ -0,0 +1,230 @@
+)
+@click.pass_context
+def gordo_client(ctx: click.Context, *args, **kwargs):
+    """Entry sub-command for client related activities."""
+    # Setup the session with any attributes set by the user
+    session_config = kwargs.pop("session_config", None)
+    if session_config:
+        session = Session()
+        for key, value in session_config.items():
+            setattr(session, key, value)
+        kwargs["session"] = session
+
+    ctx.obj = {"args": args, "kwargs": kwargs}
+
+
+@click.command("predict")
+@click.argument("start", type=IsoFormatDateTime())
+@click.argument("end", type=IsoFormatDateTime())
+@click.option(
+    "--target",
+    help="A list of machines to target. If not provided then target all machines in the project",
+    default=[],
+    multiple=True,
+)
+@click.option(
+    "--data-provider",
+    type=DataProviderParam(),
+    envvar="DATA_PROVIDER",
+    help="DataProvider dict encoded as json. Must contain a 'type' key with the name of a DataProvider as value.",
+)
+@click.option(
+    "--output-dir",
+    type=click.Path(exists=True),
+    help="Save output prediction dataframes in a directory",
+)
+@click.option(
+    "--influx-uri",
+    help="Format: <username>:<password>@<host>:<port>/<optional-path>/<db_name>",
+)
+@click.option("--influx-api-key", help="Key to provide to the destination influx")
+@click.option(
+    "--influx-recreate-db",
+    help="Recreate the desintation DB before writing",
+    is_flag=True,
+    default=False,
+)
+@click.option(
+    "--forward-resampled-sensors",
+    help="forward the resampled sensor values",
+    is_flag=True,
+    default=False,
+)
+@click.option(
+    "--n-retries",
+    help="Time client should retry failed predictions",
+    type=int,
+    default=5,
+)
+@click.option(
+    "--parquet/--no-parquet",
+    help="Use parquet serialization when sending and receiving data from server",
+    default=True,
+)
+@click.pass_context
+def predict(
+    ctx: click.Context,
+    start: datetime,
+    end: datetime,
+    target: List[str],
+    data_provider: providers.GordoBaseDataProvider,
+    output_dir: str,
+    influx_uri: str,
+    influx_api_key: str,
+    influx_recreate_db: bool,
+    forward_resampled_sensors: bool,
+    n_retries: int,
+    parquet: bool,
+):
+    """Run some predictions against the target."""
+    ctx.obj["kwargs"].update(
+        {
+            "data_provider": data_provider,
+            "forward_resampled_sensors": forward_resampled_sensors,
+            "n_retries": n_retries,
+            "use_parquet": parquet,
+        }
+    )
+
+    client = Client(*ctx.obj["args"], **ctx.obj["kwargs"])
+
+    if influx_uri is not None:
+        client.prediction_forwarder = ForwardPredictionsIntoInflux(  # type: ignore
+            destination_influx_uri=influx_uri,
+            destination_influx_api_key=influx_api_key,
+            destination_influx_recreate=influx_recreate_db,
+            n_retries=n_retries,
+        )
```

Comment:
I suppose `prediction_forwarder` is supposed to be passed as an argument, like so:

```suggestion
    if influx_uri is None:
        prediction_forwarder = None
    else:
        prediction_forwarder = ForwardPredictionsIntoInflux(
            destination_influx_uri=influx_uri,
            destination_influx_api_key=influx_api_key,
            destination_influx_recreate=influx_recreate_db,
            n_retries=n_retries,
        )

    ctx.obj["kwargs"].update(
        {
            "data_provider": data_provider,
            "forward_resampled_sensors": forward_resampled_sensors,
            "n_retries": n_retries,
            "use_parquet": parquet,
            "prediction_forwarder": prediction_forwarder,
        }
    )

    client = Client(*ctx.obj["args"], **ctx.obj["kwargs"])
```